dts/bindings/fuel-gauge/zephyr,fuel-gauge-composite.yaml

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2024 Embeint Inc
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Composite fuel-gauge constructed from analog input values
+
+compatible: "zephyr,fuel-gauge-composite"
```

Comment:
funny name... `zephyr,adc-fuel-gauge`? `fuel-gauge-adc`? `sensor-fuel-gauge`? unsure

Reply:
I wanted to avoid references to ADC's etc as that could imply that the driver itself is doing ADC operations.
It's not a sensor either, just using the sensor API to query other devices.
Happy to hear alternate names though.

Reply:
yeah not too bothered by it, just a thought

Reply:
+1, the name really confused me intially. Rather than "ADC", maybe using "analog" would be OK? `zephyr,analog-fuel-gauge`?

Reply:
I understand the naming is confusing, but I don't think `analog` is really any clearer. There is nothing inherently "analog" about this driver. The angle I was going for with `composite` is that it is a synthetic device that is constructed by combining the readings from a number of different sources and some metadata about the battery.
I think any alternate naming should reflect that.

Reply:
I also found the name slightly confusing. By composite, my first thought was it would be a fuel guage composed of several separate items, e.g. voltage metetering, current metering, etc. But instead, this is a poor man's fuel gauge. We've all been there, hehe.

So some proposals from me which I would find more intuitive (subjective, of course):

- voltage-only-fuel-gauge
- voltage-based-fuel-gauge 
- poor-man-fuel-gauge
- simple-fuel-gauge
- ocv-fuel-gauge

Anyway, happy to see this PR.